sentinel/tests/unit/transitions/accept_patient_reports.js

```diff
@@ -197,6 +197,123 @@ describe('accept_patient_reports', () => {
       });
     });
 
+    it('adds report_uuid property', done => {
+      sinon.stub(db.medic, 'post').resolves({});
+      sinon.stub(transition, '_silenceReminders').callsArgWith(3, null, true);
+      const doc = {
+        _id: 'z',
+        fields: { patient_id: 'x' },
+        reported_date: '2018-09-28T18:45:00.000Z',
+      };
+      const config = { silence_type: 'x', messages: [] };
+      const registrations = [
+        {
+          _id: 'a',
+          reported_date: '2017-02-05T09:23:07.853Z',
+          scheduled_tasks: [
+            {
+              due: '2018-09-26T18:45:00.000Z',
+              state: 'sent',
+              messages: [
+                {
+                  uuid: 'k',
+                },
+              ],
+            },
+            {
+              due: '2018-10-26T18:45:00.000Z',
+              state: 'scheduled',
+              messages: [
+                {
+                  uuid: 'j',
+                },
+              ],
+            },
+          ],
+        },
+      ];
+      sinon
+        .stub(utils, 'getRegistrations')
+        .callsArgWith(1, null, registrations);
+      transition._handleReport(doc, config, (err, complete) => {
+        complete.should.equal(true);
+        doc._id.should.equal(registrations[0].scheduled_tasks[0].report_uuid);
```

Comment:
It's confusing that this is the wrong way around, as it's implying that the code is changing the document id to the report_uuid and not the other way around. Tests are also in part documentation of expected behaviour, so it's a good idea to flip this the other way.